src/OpenSage.Game/Graphics/MeshVertex.cs

```diff
@@ -9,11 +9,16 @@ public static class MeshVertex
         [StructLayout(LayoutKind.Sequential)]
         public struct Basic
         {
-            public Vector3 Position;
-            public Vector3 Normal;
+            public Vector3 Position0;
```

Comment:
I think there are now enough skinning-specific attributes that we should have separate structs and code paths for skinning / non-skinning. But that can be done later.

Reply:
Skinned meshes in at least TW has up to two bones, so wouldn't it make sense to have this a bit more dynamic and create the VertexLayoutDescription according to what is actually there to reduce the array size?

Reply:
I think we should do that, at the same time as we allow different vertex attributes for skinning / non-skinning.

Reply:
The problem will be in keeping the number of shader permutations under control. We don’t have a good way yet of dealing with those.

Reply:
Doesn't really matter, does it? The shader uses registers which is what the VertexLayoutDescription is for.

Reply:
I assumed there would be warnings or errors if the vertex data doesn’t contain everything expected by the shader. At least, that’s how it works in D3D.

Reply:
We’d need to be able to basically compile multiple entry points per shader, one for each unique set of vertex attributes, and we can’t do that yet. It’s enough work that we will tackle it separately.

Reply:
I see, TW uses macros and shader arrays for that.